contrib/epee/include/net/abstract_tcp_server2.inl

```diff
@@ -1173,6 +1269,30 @@ POP_WARNINGS
 
     _dbg3("Connected success to " << adr << ':' << port);
 
+    const epee::net_utils::ssl_support_t ssl_support = new_connection_l->get_ssl_support();
+    if (ssl_support == epee::net_utils::e_ssl_support_enabled || ssl_support == epee::net_utils::e_ssl_support_autodetect)
+    {
+      // Handshake
+      MDEBUG("Handshaking SSL...");
+      if (!new_connection_l->handshake(boost::asio::ssl::stream_base::client))
+      {
+        if (ssl_support == epee::net_utils::e_ssl_support_autodetect)
+        {
+          MERROR("SSL handshake failed on an autodetect connection, reconnecting without SSL");
+          new_connection_l->disable_ssl();
+          boost::system::error_code ignored_ec;
+          sock_.shutdown(boost::asio::ip::tcp::socket::shutdown_both, ignored_ec);
+          sock_.close();
+          goto again;
```

Comment:
That code is like 80 lines of existing code you're asking to be moved, if I understand what you're asking, while this patch is much less intrusive. Or am I missing your point ?
Also, can you give an example of a change in the connection class which would break this ? Maybe that will let me see the point.

Reply:
There is a chunk of code that needs to be executed a maximum of twice. Instead this is creating a loop with a non-obvious termination condition which is likely harder to maintain.   I am proposing something like:

```c++
if (!try_connect(...))
{
    m_ssl_support = epee::net_utils::ssl_support_t::e_ssl_support_disabled;
    if (!try_connect(...))
        return false;
}
```
which executes a specified number of times.